Confus/Confus/Flag.cpp

```diff
@@ -8,67 +8,74 @@
 
 namespace Confus {
 
-	Flag::Flag(irr::IrrlichtDevice* a_Device, ETeamIdentifier a_TeamIdentifier) : m_TeamIdentifier(&a_TeamIdentifier) {
+	Flag::Flag(irr::IrrlichtDevice* a_Device, ETeamIdentifier a_TeamIdentifier) : m_TeamIdentifier(new ETeamIdentifier(a_TeamIdentifier)) {
```

Comment:
This seems to be a leak. Do you really need it to be a pointer in the first place or are you just trying to satisfy the techno requirements?